src/server/acl/acl_family.cc

```diff
@@ -57,10 +61,113 @@ void AclFamily::List(CmdArgList args, ConnectionContext* cntx) {
   }
 }
 
+namespace {
+
+std::optional<std::string> MaybeParsePassword(std::string_view command) {
+  if (command[0] != '>') {
+    return {};
+  }
+
+  return {std::string(command.substr(1))};
+}
+
+std::optional<bool> MaybeParseStatus(std::string_view command) {
+  if (command == "ON") {
+    return true;
+  }
+  if (command == "OFF") {
+    return false;
+  }
+  return {};
+}
+
+using OptCat = std::optional<uint32_t>;
+
+std::pair<OptCat, OptCat> MaybeParseAclCategory(std::string_view command) {
```

Comment:
I would note somewhere that the first are added cats and the second removed cats (though can be inferred from the +/- signs below)

Reply:
Actually, don't you think that returning a Cat + bool (add) would be much shorter and easier than working with the pair?

Reply:
That's a very good suggestion -- I like this!